### PR TITLE
feat(AWSS3): Update models to latest; support NSSecureCoding for most types

### PR DESCRIPTION
**Pull request by palpatim**
This PR manually updates the S3 model to the latest version, and adds support for `NSSecureCoding` to S3 model types.

The model update is manual because of hand-coded support for a `AWSS3BucketLocationConstraintBlank` value that is not automatically generated. I re-added support for that value after generating the updated AWSS3Model files.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AWSS3/AWSS3Model.h [hunk 1]
```diff
@@ -6295,14 +6956,14 @@ typedef NS_ENUM(NSInteger, AWSS3Types) {
...
- <p>The Amazon Resource Name (ARN) of the Amazon SQS queue to which Amazon S3 will publish a message when it detects events of the specified type.</p>
+ <p>The Amazon Resource Name (ARN) of the Amazon SQS queue to which Amazon S3 publishes a message when it detects events of the specified type.</p>
  */
 @property (nonatomic, strong) NSString * _Nullable queueArn;
 
 @end
 
 /**
- 
+ <p>This data type is deprecated. Use <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_QueueConfiguration.html">QueueConfiguration</a> for the same purposes. This data type specifies the configuration for publishing messages to an Amazon Simple Queue Service (Amazon SQS) queue when Amazon S3 detects specified events. </p>
```
Comment (royjit) [inline AWSS3/AWSS3Model.h:6966]: Do we add deprecated warning in codegened models? I donot think we have, just checking..

File: AWSS3UnitTests/AWSGeneralS3Tests.m [hunk 2]
```diff
@@ -626,6 +626,53 @@ - (void)testDeleteBucketMetricsConfigurationCompletionHandler {
...
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+
+	[[AWSS3 S3ForKey:key] deleteBucketOwnershipControls:[AWSS3DeleteBucketOwnershipControlsRequest new] completionHandler:^(NSError * _Nullable error) {
+        XCTAssertNotNil(error);
+        XCTAssertEqualObjects(@"OCMockExpectedNetworkingError", error.domain);
+        XCTAssertEqual(8848, error.code);
+        dispatch_semaphore_signal(semaphore);
+    }];
+	
+ 	dispatch_semaphore_wait(semaphore, dispatch_time(DISPATCH_TIME_NOW, (int)(2.0 * NSEC_PER_SEC)));
```
Comment (royjit) [inline AWSS3UnitTests/AWSGeneralS3Tests.m:670]: nit: Indentation fix

**Review by royjit — APPROVED** [review]
LGTM, some minor comments.

File: AWSS3/AWSS3Model.h (hunk 1 above)
Comment (palpatim) [reply AWSS3/AWSS3Model.h:6966]: That came from the model itself, so we just pass it through.

File: AWSS3UnitTests/AWSGeneralS3Tests.m (hunk 2 above)
Comment (palpatim) [reply AWSS3UnitTests/AWSGeneralS3Tests.m:670]: This is an artifact of code generation. I'll investigate fixing it in that process, but won't block this release for it.

**Comment by palpatim** [comment]
Refs: #2914

**Pull request merged**